src/index.ts

```diff
@@ -62,12 +73,14 @@ export default function supernova(env) {
       const changeSortOrder = useSorting(model);
 
       const [pageInfo, setPageInfo] = useState(initialPageInfo);
-      const [tableData] = usePromise(() => {
-        return env.carbon ? nothing() : manageData(model, layout, pageInfo, setPageInfo);
-      }, [layout, pageInfo]);
+      const [tableData] = usePromise(
+        async () => (env.carbon ? nothing() : manageData(model, layout, pageInfo, setPageInfo)),
+        [layout, pageInfo]
+      );
 
       useEffect(() => {
-        if (!env.carbon && reactRoot && layout && tableData && announce && changeSortOrder && theme) {
+        const isReadyToRender = !env.carbon && reactRoot && layout && tableData && announce && changeSortOrder && theme;
+        isReadyToRender &&
```

Comment:
was this to fix the problem with too many things in the if condition?

Reply:
yes

Reply:
codeclimate complains